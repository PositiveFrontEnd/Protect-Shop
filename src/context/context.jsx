import React from "react";
import { createContext, useState } from "react";

export const ContextFunctions = createContext()

const ContextProvider = ({ children }) => {
	const [isModal, setIsModal] = useState(false)

	function modalChange() {
		setIsModal(!isModal)
	}

	const [isModalAll, useIsModalAll] = useState(false)

	function modalChangeAll() {
		useIsModalAll(!isModalAll)
	}
	const [isModalAddBasket, useIsModalAddBasket] = useState(false)

	function modalChangeAddBasket() {
		useIsModalAddBasket(!isModalAddBasket)
	}

	return (
		<ContextFunctions.Provider value={{ modalChange, isModal, isModalAll, modalChangeAll, modalChangeAddBasket, isModalAddBasket }} >
			{children}
		</ContextFunctions.Provider>
	)
}

export default ContextProvider