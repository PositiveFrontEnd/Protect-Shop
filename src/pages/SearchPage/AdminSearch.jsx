import React from 'react';
import {useDispatch, useSelector} from "react-redux";
import {selectInputText, selectorSearchProducts, selectorThreeProducts, selectorToken} from "src/store/selectors";
import {actionInputText} from "src/store/homeSlice";
import {actionGetOneProduct, actionGetThreeProducts, actionSearchProducts} from "src/store/productsSlice";
// import {useNavigate} from "react-router";
import {actionFavoriteForAll} from "src/store/favoriteSlice";
import Button from "src/components/Button/Button";
import PrimaryCard from "src/components/Main/Cards/PrimaryCard/PrimaryCard/PrimaryCard";

const AdminSearch = () => {
    const inputText = useSelector(selectInputText);
    const token = useSelector(selectorToken);
    const dispatch = useDispatch();
    const handleInputChange = (event) => {
        dispatch(actionInputText(event.target.value));
    }
    const handleSearch = () => {
        dispatch(actionSearchProducts({query: `${inputText}`}))
    }
    const products = useSelector(selectorSearchProducts);
    const handleEnterPress = (event) => {
        if (event.key === 'Enter') {
            return handleSearch()
        }
    }
    const handleSubmit = (e) => {
        e.preventDefault()
    };
    // const navigate = useNavigate();
    const threeProducts = useSelector(selectorThreeProducts);
    const handleFavorite = (productId, event) => {
        event.stopPropagation();
        dispatch(actionFavoriteForAll({productId, token}));
    };

    const handleProduct = (item) => {
        dispatch(actionGetOneProduct(item._id));
        dispatch(actionGetThreeProducts(item.name));
        // navigate(
        //     `/catalogue/${item.categories}/${item.type}/${item._id}/${item.color}`
        // );
    };
    return (
        <div className='container'>
            <div className='search__input-group'>
                <form onSubmit={handleSubmit}>
                    <input value={inputText} type='text' name='search'
                           onChange={handleInputChange}
                           className='search__input'
                           onKeyDown={handleEnterPress}/>
                </form>
                <Button black onClick={handleSearch} className='search__btn'>SEARCH</Button>
            </div>
            <div className='search__result'>
                {products.map((item) => (
                    <PrimaryCard key={item._id} card={item} arr={threeProducts}
                                 handleProduct={handleProduct}
                                 handleFavorite={handleFavorite}
                    />
                ))}
            </div>
        </div>
    );
};

export default AdminSearch;