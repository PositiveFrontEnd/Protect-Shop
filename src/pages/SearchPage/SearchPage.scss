.search__title {
  padding: 8px 0;
  color: var(--black);
  font-size: 18px;
  line-height: 25px;
  text-align: center;
  text-transform: uppercase;

  @media (width >= 768px) {
    padding: 10px 0;
    font-weight: 500;
  }

  @media (width >= 1024px) {
    padding: 20px 0;
    font-size: 24px;
    line-height: 33px;
  }

  span {
    display: none;

    @media (width >= 768px) {
      display: inline;
      margin-left: 10px;
    }
  }
}

.search__block {
  @media (width >= 768px) {
    display: flex;
    gap: 6px;
    align-items: start;
  }
}

.search__input-group {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin-top: 8px;
  text-transform: uppercase;
  border: 1px solid var(--black);
  border-radius: 12px;

  @media (width >= 768px) {
    order: 2;
    height: 48px;
  }

  .search__input {
    padding-left: 16px;
    font-size: 16px;
    line-height: 22px;
  }
}

.search__btn {
  max-width: 170px;
  font-size: 12px;
  border-radius: 0 12px 12px 0;

  @media (width >= 768px) {
    height: 48px;
    padding: 13px 0;
    max-width: 200px;
    font-size: 14px;
  }
  @media (width >=1024px) {
    max-width: 240px;
    font-size: 16px;
  }
}

.search__function-group {
  display: flex;
  gap: 8px;
  align-items: start;
  justify-content: space-between;
  width: 100%;
  margin-top: 16px;

  @media (width >= 768px) {
    order: 1;
    width: 160px;
    margin-top: 8px;
  }

  @media (width >= 1024px) {
    width: 311px;
  }
}

.search__result__title {
  margin-top: 24px;
  line-height: 22px;
  text-align: center;
  text-transform: uppercase;

  @media (width >= 768px) {
    margin-top: 30px;
  }

  @media (width >= 1024px) {
    margin-top: 62px;
    font-size: 18px;
  }
  span {
    text-decoration: underline;
  }
}

.search__result {
  display: grid;
  grid-template-columns: 1fr 1fr;
  row-gap: 30px;
  column-gap: 12px;
  justify-content: center;
  margin-top: 28px;

  @media (width >= 768px) {
    grid-template-columns: repeat(3, 1fr);
    margin-top: 50px;
  }

  @media (width >= 1024px) {
    grid-template-columns: repeat(4, 1fr);
    margin-top: 80px;
  }
}

.button__SeeMore-white {
  margin: 40px auto 0;
}

.button-inactive {
  display: none;
}

.sort {
  display: flex;
  justify-content: space-between;
  width: 100%;
  min-width: 160px;
  padding: 8px 16px;
  font-size: 12px;
  line-height: 16px;
  text-transform: uppercase;
  border: 1px solid var(--black);
  border-radius: 12px;

  @media (width >= 768px) {
    height: 48px;
    padding: 14px 16px;
    font-weight: 600;
  }

  @media (width >= 1024px) {
    width: 311px;
    font-size: 16px;
  }
}

.sort-open {
  display: flex;
  justify-content: space-between;
  width: 160px;
  padding: 8px 16px;
  font-size: 12px;
  line-height: 16px;
  text-transform: uppercase;
  background: var(--inactive);
  border: 1px solid var(--black);
  border-bottom: none;
  border-radius: 12px 12px 0 0;

  @media (width >= 768px) {
    height: 48px;
    padding: 14px 16px;
    font-weight: 600;
  }

  @media (width >= 1024px) {
    width: 311px;
    font-size: 16px;
  }
}

.sort__dropdown {
  display: flex;
  flex-direction: column;
  gap: 4px;
  padding: 8px 16px;
  font-size: 14px;
  line-height: 19px;
  border: 1px solid var(--black);
  border-top: none;
  border-radius: 0 0 12px 12px;

  @media (width >= 1024px) {
    width: 311px;
  }
}

.sort__dropdown input {
  display: none;
}

.sort__dropdown label:hover {
  font-weight: 600;
}
