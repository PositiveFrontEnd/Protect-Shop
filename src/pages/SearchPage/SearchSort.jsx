import React, { useState } from 'react';
import ArrowDown from './SearchSvg/arrowDown.svg?react';
import ArrowUp from './SearchSvg/arrowUp.svg?react';
import PropTypes from "prop-types";

const Sort = ({ handleSort, sort }) => {
    const [isOpenSort, setIsOpenSort] = useState(false);
    const actionSort = () => {
        setIsOpenSort(!isOpenSort)
    };
    return (
        <div>
            <button className={isOpenSort ? 'sort-open' : 'sort'}
                onClick={actionSort}>sort
                {
                    isOpenSort ? <ArrowUp /> : <ArrowDown />
                }
            </button>
            {isOpenSort && (
                <div className='sort__dropdown'>
                    <label><input type='radio'
                        value='lowToHigh'
                        checked={sort === 'lowToHigh'}
                        onChange={handleSort} />
                        Price low to high</label>
                    <label><input type='radio'
                        value='highToLow'
                        checked={sort === 'highToLow'}
                        onChange={handleSort} />
                        Price high to low</label>
                </div>
            )}
        </div>
    );
};
Sort.propTypes = {
    handleSort: PropTypes.func,
    sort: PropTypes.string
}
export default Sort;