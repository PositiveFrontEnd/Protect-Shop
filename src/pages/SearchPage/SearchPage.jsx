import React from 'react';
import './SearchPage.scss'
import Hero from "src/components/Main/Hero/Hero";
import Instagram from "src/components/Main/Instagram/Instagram";
import HeaderSearch from "src/components/Header/HeaderSearch";
import Blog from "src/components/Main/Blog/Blog";
import Feedbacks from "src/components/Main/Feedbacks/Feedbacks";
import OurProducts from "src/components/Main/OurProducts/OurProducts";
import Popular from "src/components/Main/Popular/Popular";

const SearchPage = () => {
    return (
        <div>
            <HeaderSearch />
            <Hero />
            <OurProducts />
            <Popular />
            <Feedbacks />
            <Blog />
            <Instagram />
        </div>
    );
};

export default SearchPage;