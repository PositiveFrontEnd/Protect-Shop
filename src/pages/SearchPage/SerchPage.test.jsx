import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import SearchPage from "./SearchPage";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { ContextFunctions } from "../../context/context";
import thunk from "redux-thunk";
describe("Тестування компонента  SearchPage", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {
            home: {
                isAnimation: false,
                inputText: "some text",
                filterHomePage: [],
            },

            user: {
                token: "token",
                registrationModal: false,
                authorizationStatus: null,
                registrationStatus: "some text",
                registrationData: {},
                correctData: "some text",
                isAdmin: false,
            },
            products: {
                products: [],
                productsByCategory: [],
                productsByType: [],
                oneProduct: {},
                ThreeProducts: [],
                ThreeColors: [],
                filters: [],
                twelveProducts: [],
                filtersParameters: {},
                twelveFiltersProducts: [],
                searchProducts: [],
                youSee: [],
                previewProductInfo: {},
            },
            order: {
                orderState: [],
                productsForOrderGuest: {},
                infoForOrderGuest: [],
                deliveryForOrderGuest: {},
                order: {
                    letterSubject: "Thank you for order!"
                },
                promoCodePrice: "",
                orderGuest: { letterSubject: "Thank you for order!" },
            },
            basket: {
                basket: [],
                priceGuest: "''",
                guestBasket: {},
            },
            catalogue: {
                sortCatalogue: "some text",
                loadFilter: false,
            },
            favorite: {
                favorite: [],
                favoriteForCustomer: [],
                guestFavorite: [],
            },
            message: {
                letters: [],
            }

        };
        const contextValues = {
            isModalAll: true,
            modalChangeAll: true,
        };

        const store = mockStore(initialState);
        const searchPage = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ContextFunctions.Provider value={contextValues}>
                        <SearchPage />
                    </ContextFunctions.Provider>
                </BrowserRouter>
            </Provider>
        );

        expect(searchPage).toMatchSnapshot();
    });
});
