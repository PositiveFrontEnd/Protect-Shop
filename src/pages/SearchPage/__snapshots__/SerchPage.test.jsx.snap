// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  SearchPage > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="header__search"
        >
          <svg
            fill="none"
            height="33"
            viewBox="0 0 32 33"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22.1553 24.5414L27.8647 30.2521L29.7513 28.3654L24.0407 22.6561C25.7436 20.5317 26.6699 17.8894 26.666 15.1667C26.666 8.54275 21.29 3.16675 14.666 3.16675C8.04202 3.16675 2.66602 8.54275 2.66602 15.1667C2.66602 21.7907 8.04202 27.1667 14.666 27.1667C17.3887 27.1706 20.031 26.2444 22.1553 24.5414ZM21.166 21.8667C19.4259 23.5589 17.0933 24.5039 14.666 24.5001C9.50868 24.5001 5.33268 20.3227 5.33268 15.1667C5.33268 10.0094 9.50868 5.83341 14.666 5.83341C19.822 5.83341 23.9993 10.0094 23.9993 15.1667C24.0032 17.594 23.0582 19.9266 21.366 21.6667L21.166 21.8667Z"
              fill="#191919"
            />
          </svg>
          <form>
            <input
              class="header__search__title"
              name="search"
              placeholder="Search"
              type="text"
              value=""
            />
          </form>
        </div>
        <section
          class="hero__wrapper"
        >
          <div
            class="swiper swiper-initialized swiper-horizontal mySwiper"
          >
            <div
              class="swiper-wrapper"
              style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
            >
              <div
                class="swiper-slide hero__img__wrapper"
                data-swiper-slide-index="0"
              >
                <div>
                  <img
                    src="/Images/heroSale.jpg"
                  />
                  <div
                    class="title__button__wrapper"
                  >
                    <p
                      class="title"
                    >
                      Hurry up to get highest discount
                    </p>
                    <a
                      class="button-black hero__button"
                      href="/catalogue?status=sell"
                    >
                        see more
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="swiper-slide hero__img__wrapper"
                data-swiper-slide-index="1"
              >
                <div>
                  <img
                    src="/Images/heroNewCollection.jpg"
                  />
                  <div
                    class="title__button__wrapper"
                  >
                    <p
                      class="title"
                    >
                      New collection for summer travel
                    </p>
                    <a
                      class="button-black hero__button"
                      href="/catalogue?categories=men,woman&type=backpack"
                    >
                        see more
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="swiper-slide hero__img__wrapper"
                data-swiper-slide-index="2"
              >
                <div>
                  <img
                    src="/Images/heroAdventure.jpg"
                  />
                  <div
                    class="title__button__wrapper"
                  >
                    <p
                      class="title"
                    >
                      Essentials for Every Adventure
                    </p>
                    <a
                      class="button-black hero__button"
                      href="/catalogue?categories=children&type=backpack"
                    >
                        see more
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="swiper-button-prev"
            />
            <div
              class="swiper-button-next"
            />
            <div
              class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
            >
              <span
                class="swiper-pagination-bullet swiper-pagination-bullet-active"
              />
              <span
                class="swiper-pagination-bullet"
              />
              <span
                class="swiper-pagination-bullet"
              />
            </div>
          </div>
        </section>
        <section>
          <div
            class="products__wrapper"
          >
            <div
              class="container products"
            >
              <div
                class="products__images"
              >
                <div
                  class="products__images"
                >
                  <img
                    alt=""
                    src="/Images/forWomen.jpg"
                  />
                  <a
                    class="button-black products__image__caption"
                    href="/catalogue?categories=woman"
                  >
                    for woman
                  </a>
                </div>
                <div
                  class="products__images"
                >
                  <img
                    alt=""
                    src="/Images/forKids.jpg"
                  />
                  <a
                    class="button-black products__image__caption"
                    href="/catalogue?categories=children"
                  >
                    for children
                  </a>
                </div>
                <div
                  class="products__images"
                >
                  <img
                    alt=""
                    src="/Images/forMen.jpg"
                  />
                  <a
                    class="button-black products__image__caption"
                    href="/catalogue?categories=men"
                  >
                    for men
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="popular__conteiner"
        >
          <div
            class="container"
          >
            <h4
              class="popular__title"
            >
              The Most Popular shopping for her
            </h4>
            <div
              class="popular__wrapper"
            />
          </div>
        </section>
        <div
          class="special-offer__wrapper"
        >
          <div
            class="container buttons"
          >
            <button
              class="button-black special-offer__button"
              type="button"
            >
              New here $20 off your first purchase
            </button>
            <button
              class="button-black special-offer__button"
              type="button"
            >
              Get a 10% discount for signing up
            </button>
          </div>
          <div
            class="modal__wrapper"
          >
            <div
              class="modal"
            >
              <div
                class="modal__header"
              >
                <div
                  class="header__title"
                >
                  <svg
                    class="header__svg"
                    fill="none"
                    viewBox="0 0 32 32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M27.1129 6.00184C26.4113 5.26384 25.5756 4.67847 24.6544 4.27979C23.7333 3.88113 22.745 3.67713 21.7474 3.67971C20.7497 3.67713 19.7616 3.88113 18.8404 4.27979C17.9193 4.67847 17.0836 5.26384 16.3819 6.00184L15.9539 6.46309L15.5259 6.01773C14.1069 4.54144 12.1824 3.71208 10.1757 3.71208C8.169 3.71208 6.2445 4.54144 4.82548 6.01773C3.42504 7.50429 2.64062 9.5029 2.64062 11.5845C2.64062 13.6661 3.42504 15.6647 4.82548 17.1512L15.1743 27.9507C15.3892 28.1742 15.6806 28.2997 15.9845 28.2997C16.2882 28.2997 16.5798 28.1742 16.7947 27.9507L27.1435 17.1512C28.5445 15.6598 29.3262 13.6555 29.3206 11.57C29.3149 9.48458 28.522 7.48492 27.1129 6.00184Z"
                      fill="#FEFEFE"
                    />
                  </svg>
                  <p
                    class="title__text"
                  >
                    Promo code
                  </p>
                </div>
                <button
                  class="header__button"
                  type="button"
                >
                  <svg
                    fill="black"
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                      fill="white"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="modal__container"
              >
                <div
                  class="modal__content modal__content-form"
                >
                  <p
                    class="modal__body-title title__logOut"
                  >
                    YOUR PROMO CODE: NEW1023
                  </p>
                  <div
                    class="footer__email"
                  >
                    <button
                      class="button-white modal__button"
                      type="button"
                    >
                      View my orders
                    </button>
                    <button
                      class="button-black modal__button"
                      type="button"
                    >
                      Continue shopping
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <section>
          <div
            class="feedbacks__wrapper"
          >
            <div
              class="container feedbacks"
            >
              <h2
                class="feedbacks__title"
              >
                feedbacks
              </h2>
              <div
                class="feedbacks__content"
              />
              <div
                class="feedbacks__button"
              >
                <button
                  class="button-white"
                  type="button"
                >
                  see more
                </button>
              </div>
            </div>
          </div>
        </section>
        <section
          class="blog__wrapper"
        >
          <div
            class="container"
          >
            <h2
              class="blog__title"
            >
              blog
            </h2>
            <ul
              class="blog__info"
            >
              <li
                class="mobile__description"
              >
                <img
                  alt="sectionBlogOne"
                  class="blog__img"
                  src="/Images/sectionBlogOne.jpg"
                />
                <h4
                  class="blog__info__title"
                >
                  Mini bags forever 
                </h4>
                <p
                  class="blog__info__description"
                >
                  Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
                </p>
              </li>
              <li
                class="develop__description"
              >
                <img
                  alt=""
                  class="blog__img"
                  src="/Images/sectionBlogTwo.jpg"
                />
                <h4
                  class="blog__info__title"
                >
                  Mini bags forever 
                </h4>
                <p
                  class="blog__info__description"
                >
                  Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
                </p>
              </li>
              <li
                class="develop__description"
              >
                <img
                  alt=""
                  class="blog__img"
                  src="/Images/sectionBlogThree.jpg"
                />
                <h4
                  class="blog__info__title"
                >
                  Sackbags
                </h4>
                <p
                  class="blog__info__description"
                >
                  Sackbags with large fasteners and chains have become popular this season.
                </p>
                <p
                  class="blog__info__description"
                >
                  These bags are suitable for many styles, from classics to modern minimalism.
                </p>
              </li>
            </ul>
            <div
              class="blog__button"
            >
              <button
                class="button-white"
                type="button"
              >
                See more
              </button>
            </div>
          </div>
        </section>
        <section
          class="insta__wrapper "
        >
          <div
            class="container"
          >
            <h4
              class="insta__title"
            >
              We are in Instagram @Protect.com
            </h4>
            <div
              class="insta__img__wrapper"
            />
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="header__search"
      >
        <svg
          fill="none"
          height="33"
          viewBox="0 0 32 33"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22.1553 24.5414L27.8647 30.2521L29.7513 28.3654L24.0407 22.6561C25.7436 20.5317 26.6699 17.8894 26.666 15.1667C26.666 8.54275 21.29 3.16675 14.666 3.16675C8.04202 3.16675 2.66602 8.54275 2.66602 15.1667C2.66602 21.7907 8.04202 27.1667 14.666 27.1667C17.3887 27.1706 20.031 26.2444 22.1553 24.5414ZM21.166 21.8667C19.4259 23.5589 17.0933 24.5039 14.666 24.5001C9.50868 24.5001 5.33268 20.3227 5.33268 15.1667C5.33268 10.0094 9.50868 5.83341 14.666 5.83341C19.822 5.83341 23.9993 10.0094 23.9993 15.1667C24.0032 17.594 23.0582 19.9266 21.366 21.6667L21.166 21.8667Z"
            fill="#191919"
          />
        </svg>
        <form>
          <input
            class="header__search__title"
            name="search"
            placeholder="Search"
            type="text"
            value=""
          />
        </form>
      </div>
      <section
        class="hero__wrapper"
      >
        <div
          class="swiper swiper-initialized swiper-horizontal mySwiper"
        >
          <div
            class="swiper-wrapper"
            style="transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="swiper-slide hero__img__wrapper"
              data-swiper-slide-index="0"
            >
              <div>
                <img
                  src="/Images/heroSale.jpg"
                />
                <div
                  class="title__button__wrapper"
                >
                  <p
                    class="title"
                  >
                    Hurry up to get highest discount
                  </p>
                  <a
                    class="button-black hero__button"
                    href="/catalogue?status=sell"
                  >
                      see more
                  </a>
                </div>
              </div>
            </div>
            <div
              class="swiper-slide hero__img__wrapper"
              data-swiper-slide-index="1"
            >
              <div>
                <img
                  src="/Images/heroNewCollection.jpg"
                />
                <div
                  class="title__button__wrapper"
                >
                  <p
                    class="title"
                  >
                    New collection for summer travel
                  </p>
                  <a
                    class="button-black hero__button"
                    href="/catalogue?categories=men,woman&type=backpack"
                  >
                      see more
                  </a>
                </div>
              </div>
            </div>
            <div
              class="swiper-slide hero__img__wrapper"
              data-swiper-slide-index="2"
            >
              <div>
                <img
                  src="/Images/heroAdventure.jpg"
                />
                <div
                  class="title__button__wrapper"
                >
                  <p
                    class="title"
                  >
                    Essentials for Every Adventure
                  </p>
                  <a
                    class="button-black hero__button"
                    href="/catalogue?categories=children&type=backpack"
                  >
                      see more
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="swiper-button-prev"
          />
          <div
            class="swiper-button-next"
          />
          <div
            class="swiper-pagination swiper-pagination-clickable swiper-pagination-bullets swiper-pagination-horizontal"
          >
            <span
              class="swiper-pagination-bullet swiper-pagination-bullet-active"
            />
            <span
              class="swiper-pagination-bullet"
            />
            <span
              class="swiper-pagination-bullet"
            />
          </div>
        </div>
      </section>
      <section>
        <div
          class="products__wrapper"
        >
          <div
            class="container products"
          >
            <div
              class="products__images"
            >
              <div
                class="products__images"
              >
                <img
                  alt=""
                  src="/Images/forWomen.jpg"
                />
                <a
                  class="button-black products__image__caption"
                  href="/catalogue?categories=woman"
                >
                  for woman
                </a>
              </div>
              <div
                class="products__images"
              >
                <img
                  alt=""
                  src="/Images/forKids.jpg"
                />
                <a
                  class="button-black products__image__caption"
                  href="/catalogue?categories=children"
                >
                  for children
                </a>
              </div>
              <div
                class="products__images"
              >
                <img
                  alt=""
                  src="/Images/forMen.jpg"
                />
                <a
                  class="button-black products__image__caption"
                  href="/catalogue?categories=men"
                >
                  for men
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="popular__conteiner"
      >
        <div
          class="container"
        >
          <h4
            class="popular__title"
          >
            The Most Popular shopping for her
          </h4>
          <div
            class="popular__wrapper"
          />
        </div>
      </section>
      <div
        class="special-offer__wrapper"
      >
        <div
          class="container buttons"
        >
          <button
            class="button-black special-offer__button"
            type="button"
          >
            New here $20 off your first purchase
          </button>
          <button
            class="button-black special-offer__button"
            type="button"
          >
            Get a 10% discount for signing up
          </button>
        </div>
        <div
          class="modal__wrapper"
        >
          <div
            class="modal"
          >
            <div
              class="modal__header"
            >
              <div
                class="header__title"
              >
                <svg
                  class="header__svg"
                  fill="none"
                  viewBox="0 0 32 32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M27.1129 6.00184C26.4113 5.26384 25.5756 4.67847 24.6544 4.27979C23.7333 3.88113 22.745 3.67713 21.7474 3.67971C20.7497 3.67713 19.7616 3.88113 18.8404 4.27979C17.9193 4.67847 17.0836 5.26384 16.3819 6.00184L15.9539 6.46309L15.5259 6.01773C14.1069 4.54144 12.1824 3.71208 10.1757 3.71208C8.169 3.71208 6.2445 4.54144 4.82548 6.01773C3.42504 7.50429 2.64062 9.5029 2.64062 11.5845C2.64062 13.6661 3.42504 15.6647 4.82548 17.1512L15.1743 27.9507C15.3892 28.1742 15.6806 28.2997 15.9845 28.2997C16.2882 28.2997 16.5798 28.1742 16.7947 27.9507L27.1435 17.1512C28.5445 15.6598 29.3262 13.6555 29.3206 11.57C29.3149 9.48458 28.522 7.48492 27.1129 6.00184Z"
                    fill="#FEFEFE"
                  />
                </svg>
                <p
                  class="title__text"
                >
                  Promo code
                </p>
              </div>
              <button
                class="header__button"
                type="button"
              >
                <svg
                  fill="black"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                    fill="white"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
            <div
              class="modal__container"
            >
              <div
                class="modal__content modal__content-form"
              >
                <p
                  class="modal__body-title title__logOut"
                >
                  YOUR PROMO CODE: NEW1023
                </p>
                <div
                  class="footer__email"
                >
                  <button
                    class="button-white modal__button"
                    type="button"
                  >
                    View my orders
                  </button>
                  <button
                    class="button-black modal__button"
                    type="button"
                  >
                    Continue shopping
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <section>
        <div
          class="feedbacks__wrapper"
        >
          <div
            class="container feedbacks"
          >
            <h2
              class="feedbacks__title"
            >
              feedbacks
            </h2>
            <div
              class="feedbacks__content"
            />
            <div
              class="feedbacks__button"
            >
              <button
                class="button-white"
                type="button"
              >
                see more
              </button>
            </div>
          </div>
        </div>
      </section>
      <section
        class="blog__wrapper"
      >
        <div
          class="container"
        >
          <h2
            class="blog__title"
          >
            blog
          </h2>
          <ul
            class="blog__info"
          >
            <li
              class="mobile__description"
            >
              <img
                alt="sectionBlogOne"
                class="blog__img"
                src="/Images/sectionBlogOne.jpg"
              />
              <h4
                class="blog__info__title"
              >
                Mini bags forever 
              </h4>
              <p
                class="blog__info__description"
              >
                Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
              </p>
            </li>
            <li
              class="develop__description"
            >
              <img
                alt=""
                class="blog__img"
                src="/Images/sectionBlogTwo.jpg"
              />
              <h4
                class="blog__info__title"
              >
                Mini bags forever 
              </h4>
              <p
                class="blog__info__description"
              >
                Small bags have become very popular this season. They are perfect for parties and weekends when you don't have to wear many things. Such bags can be worn both on the shoulder and at the fingertips.
              </p>
            </li>
            <li
              class="develop__description"
            >
              <img
                alt=""
                class="blog__img"
                src="/Images/sectionBlogThree.jpg"
              />
              <h4
                class="blog__info__title"
              >
                Sackbags
              </h4>
              <p
                class="blog__info__description"
              >
                Sackbags with large fasteners and chains have become popular this season.
              </p>
              <p
                class="blog__info__description"
              >
                These bags are suitable for many styles, from classics to modern minimalism.
              </p>
            </li>
          </ul>
          <div
            class="blog__button"
          >
            <button
              class="button-white"
              type="button"
            >
              See more
            </button>
          </div>
        </div>
      </section>
      <section
        class="insta__wrapper "
      >
        <div
          class="container"
        >
          <h4
            class="insta__title"
          >
            We are in Instagram @Protect.com
          </h4>
          <div
            class="insta__img__wrapper"
          />
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
